test/functional/editions_controller_test.rb

```diff
@@ -264,6 +264,46 @@ class EditionsControllerTest < ActionController::TestCase
     end
   end
 
+  context "#duplicate" do
+    setup do
+      @answer = FactoryBot.create(:answer_edition, panopticon_id: FactoryBot.create(:artefact).id)
+    end
+
+    should "redirect to edit tab and show success message when user saves successfully" do
+      EditionDuplicator.any_instance.expects(:duplicate).returns(true)
+      EditionDuplicator.any_instance.expects(:new_edition).returns(@answer)
+      post :duplicate, params: { id: @edition.id, to: "help_page" }
+
+      assert_response :found
+      assert_equal "New edition created", flash[:success]
+    end
+
+    should "render the edit tab and show a failure message when saving the new edition fails" do
+      EditionDuplicator.any_instance.expects(:duplicate).returns(false)
+      post :duplicate, params: { id: @edition.id, to: "help_page" }
+
+      assert_template "show"
+      assert_equal "Failed to create new edition: couldn't initialise", flash[:danger]
+    end
+
+    should "render the edit tab and show a failure message when another user has already created a new edition" do
+      FactoryBot.create(:edition, panopticon_id: @edition.artefact.id)
+      post :duplicate, params: { id: @edition.id, to: "help_page" }
+
+      assert_template "show"
+      assert_equal "Another person has created a newer edition", flash[:warning]
+    end
+
+    should "render the edit tab and show a failure message when there is a service problem" do
+      EditionDuplicator.any_instance.expects(:duplicate).returns(true)
```

Comment:
I think this test is passing for the wrong reason. The line below doesn't do anything, as `Edition` doesn't have a `duplicate` method (and if you comment it out, the test still passes). Should this line be stubbing an exception raise instead of returning true?
```suggestion
      EditionDuplicator.any_instance.expects(:duplicate).raises(StandardError)
```